modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/util/TypeUtils.java

```diff
@@ -379,4 +381,16 @@ private static long fromLocalTs(DataContext ctx, long ts) {
         // Taking into account DST, offset can be changed after converting from UTC to time-zone.
         return ts - tz.getOffset(ts - tz.getOffset(ts));
     }
+
+    /**
+     * @return RexLiteral
+     */
+    public static RexNode toRexLiteral(Object dfltVal, IgniteTypeFactory typeFactory, RexBuilder rexBuilder,
+        ColumnDescriptor desc) {
+        if (dfltVal instanceof UUID)
+            return rexBuilder.makeCast(typeFactory.createUUIDType(),
```

Comment:
`desc.logicalType(typeFactory)` - is already `UuidType`. We should make `varchar` literal and than cast it to uuid.

Reply:
Fixed